src/middlewares/checkRole.ts

```diff
@@ -19,7 +19,18 @@ export const checkRole = (roles: Array<string>) => {
     }
 
     //Check if array of authorized roles includes the user's role
-    if (roles.indexOf(user.role) > -1) next();
+    let rolefound = false;
+    for (let i = 0; i < roles.length; i++) {
+      const r = roles[i];
+      for (let j = 0; j < user.roles.length; j++) {
+        const ur = user.roles[j];
+        if(ur == r){
+          rolefound = true;
+          break;
+        }
+      }
+    }
+    if (rolefound) next();
```

Comment:
er der en grund til at smide det i variable frem for bare at bruge [i] og [j]? Og er det ikke muligt at finde den her info med .some?